motion/joystick_guidance/scripts/joystick_guidance.py

```diff
@@ -18,13 +55,35 @@ def __init__(self):
 
 		self.sub = rospy.Subscriber('/guidance/joystick_data', Joy, self.callback, queue_size=1)
 		self.pub = rospy.Publisher('/auv/thruster_manager/input', Wrench, queue_size=1)
-
+		self.pub_joy = rospy.Publisher('/guidance/joystick_reference', Pose, queue_size=1)
+		
 		self.surge 	= 0
 		self.sway 	= 1
 		self.heave 	= 2
-		self.roll 	= 3
-		self.pitch 	= 4
-		self.yaw 	= 5
+		self.roll 	= 3 
+		self.pitch 	= 4 
+		self.yaw 	= 5 
+
+		self.min_point_range = 0.2                          # Minimum allowed distance from the UUV to the point
+		self.max_point_range = 1                            # Maximum allowed distance from the UUV to the calculated point
+		self.num_ranges = 5                                 # Number of ranges the input is scaled by
+		self.joystick_num_bit = 16                          # Resolution on the joystick
+		self.deadzone = 1000								# Deadzone on upper input
+
+		# Required input-values to generate valid signals
+		limits = []
+		self.max_value = pow(2, self.joystick_num_bit - 1)  # -1 due to only positive integers
+		for i in range(1, self.num_ranges + 1):         
+			limits.append((self.max_value / self.num_ranges) * i)
+		self.limits = limits
+
+
+		# Calculated input in decimal
+		decimal_list = []
+		for i in range(0, self.num_ranges + 1):
+			decimal_list.append((self.max_point_range / self.num_ranges) * i)
+		self.decimal_list = decimal_list
+
```

Comment:
Here is the bug!

In python 2.7 decimal_list is set to [0, 0, 0, 0, 0, 0], while in python 3.7 it is correctly set to [0, 0.2, 0.4, 0.6, 0.8, 1]